diff --git a/node_modules/react-native-reanimated/ios/LayoutReanimation/REAAnimationsManager.m b/node_modules/react-native-reanimated/ios/LayoutReanimation/REAAnimationsManager.m
index 337b1be..62f3028 100644
--- a/node_modules/react-native-reanimated/ios/LayoutReanimation/REAAnimationsManager.m
+++ b/node_modules/react-native-reanimated/ios/LayoutReanimation/REAAnimationsManager.m
@@ -247,6 +247,7 @@ - (void)onViewRemoval:(UIView *)view before:(REASnapshot *)before
 
 - (void)onViewCreate:(UIView *)view after:(REASnapshot *)after
 {
+  _reaUiManager.flushUiOperations();
   NSNumber *tag = view.reactTag;
   if (_states[tag] == nil) {
     _states[tag] = [NSNumber numberWithInt:Inactive];
diff --git a/node_modules/react-native-reanimated/ios/LayoutReanimation/REAUIManager.mm b/node_modules/react-native-reanimated/ios/LayoutReanimation/REAUIManager.mm
index 252f263..17eee20 100644
--- a/node_modules/react-native-reanimated/ios/LayoutReanimation/REAUIManager.mm
+++ b/node_modules/react-native-reanimated/ios/LayoutReanimation/REAUIManager.mm
@@ -339,10 +339,4 @@ - (void)unregisterView:(id<RCTComponent>) view
   [viewRegistry removeObjectForKey:view.reactTag];
 }
 
-- (void)batchDidComplete
-{
-  self.flushUiOperations();
-  [super batchDidComplete];
-}
-
 @end
